### ISSUE TITLE
TypeError: an integer is required (got type NoneType)

### ISSUE THREAD

**Issue by niloofar17** [AtmaHou/FewShotJoint#1]
Hi ,

I was running the `snips_JointProto_5.sh` but I am getting the error below: 

<img width="1017" alt="Screenshot 2021-11-04 at 07 29 13" src="https://user-images.githubusercontent.com/76474413/140268052-82304746-ef36-4563-85ef-7c410a3ecea3.png">

I have noticed that in the `class FewShotSupportFeature()` in lines 144, and 145 the test_input and test feature_item are used. but I have changed it but it didn't work. I  also have compared the  `preprocessor.py` here and the one in https://github.com/laiyongkui1997/FewJoint/blob/master/utils/preprocessor.py (which I ran previously and was working fine), but there were some additional functions for `maml` which I don't need it right now and also test_input and test feature item. Do you know how can I solve this problem?

the code on the left is yours and the one on the right is for @laiyongkui1997 
<img width="1397" alt="Screenshot 2021-11-04 at 08 13 01" src="https://user-images.githubusercontent.com/76474413/140272287-bdcbd078-1c54-443e-97a7-fa67c12d81c9.png">

-there is also another request, I couldn't find the tools for generating the few-shot dataset, there was only one sample in data. As it has a different setup from the tools mentioned in https://github.com/AtmaHou/MetaDialog#few-shot-data-construction-tool , I would be more than happy if you can provide it then I can create it. 

thank you in advance for all your help and understanding.

**Comment by niloofar17** [comment]
Hi @AtmaHou , it is just a kind reminder about the problem. thank you for your help

**Comment by XiaoFengbing** [comment]
Hi, Could you provide more details about your modification/error, including how to use our project? Because I use `snips_JointProto_5.sh` to try to reproduce your error, and try to output the value of `feature_item` at the location where you reported the error, but there is no output. Did you modify something that caused the problem?

In addition, the right side of the picture is this project, not the left side, I think. Let me once again be confused about your modification.

Similarly, about the few-shot data generation tool, the data that this project provide in the `data` folder are all generated by this tool that you mentioned. I think you should further clarify your confusion.

I‘m very glad to solve your problems and confusions!

**Comment by niloofar17** [comment]
Hi @XiaoFengbing,
 thank you for your answer, sorry for switching left and right😁, I didn't change anything except the `from torchcrf import CRF` --> from TorchCRF import CRF in the `seq_laberler.py` and then I set the path to the file and  uncased_bert_model and ran the model with the same config but I got the error which I have attached in the first comment, the error `TypeError: an integer is required (got type NoneType) #1`. after that I started to check with the other repository file and I have found some differences, I applied them but still got the error. 

and about the few_shot  according to the paper: 
In Few_shot joint learning ID and SF paper it is mentioned that you used the setting of (Krone et.al. 2020a) for snips but I couldn't find any script for that in the repository. If you are following the MetaDialog tools for generating the few-shot dataset, the gen_config.py is only for setting one domain for test and one domain for dev. but in the paper, it is mentioned 2 intent for each test and dev and 3 for training. 

It was also mentioned that " For Snips, we construct 200 few-shot episodes for training, 50 for developing, and 50 for testing. We set the query set size as 16 for training and developing, 100 for testing." but in the `gen_meta_data.sh `there is no parameter to set query/dev episode number, and there only one episode_num which refers to all the 

best,
Niloofar
